.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,55 @@
+name: Unit test
+
+on:
+  release:
+    types: [created]
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "**.go"
+      - "!e2e/**" # exclude files from e2e tests
+      - ".github/workflows/unit-tests.yaml"
+
+env:
+  GO111MODULE: on
+  GOFLAGS: -mod=vendor
+
+jobs:
+  test-unit:
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+    
+    name: unit-test-${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.17
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+
+      - name: Test
+        run: ./hack/coverage.bash
+
+  test-unit-windows:
+    if: github.ref != 'refs/heads/master'
```

Comment:
why does we have this for windows but not for the others?

Reply:
I guess this is a mistake. We don't need any condition here.